ow_power_system/src/PowerSystemNode.cpp

```diff
@@ -263,30 +268,155 @@ void PowerSystemNode::injectFault (const string& fault_name,
+      {
+        // It's a different fault profile from whatever the previous stored directory was, OR the
+        // previous fault ran its course, so reset index.      
+        // Load a new fault profile from the provided directory.
+        try
+        {
+          ifstream file(current_fault_directory);
+          if (current_fault_directory.substr(current_fault_directory.size() - 4) != ".csv")
+          {
+            throw -1;
+          }
+          else if (file.fail())
+          {
+            throw -2;
+          }
+          if (end_fault_warning_displayed && designated_file == saved_file)
+          {
+            ROS_INFO_STREAM("Reloading " << designated_file << "...");
+          }
+          else if (saved_fault_directory != "N/A")
+          {
+            ROS_INFO_STREAM("Loading " << designated_file << " and unloading " << saved_file << "...");
+          }
+          else
+          {
+            ROS_INFO_STREAM("Loading " << designated_file << "...");
+          }
+          loadCustomFaultPowerProfile(current_fault_directory);
+          custom_fault_ready = true;
+          end_fault_warning_displayed = false;
+          // TEMP: Uncomment this and the other 'TEMP' line code to re-enable restricting power faults:
+          // When uncommented, the user will only be able to use either high power draw or the custom
+          // fault feature, never both in a single simulation. Behavior of both at once is unpredictable.
+          //CUSTOM_FAULT_ACTIVATED = true;
+          saved_fault_directory = current_fault_directory;
+          saved_file = designated_file;
+          index = 0;
+
+          ROS_WARN_STREAM_ONCE("Custom power faults may exhibit unexpected results. Caution is advised.");
+          //ROS_WARN_STREAM_ONCE("Note that custom power faults cannot be combined with high power draw.");
+          ROS_INFO_STREAM(saved_file.substr(0, saved_file.size() - 4) << " activated!");
+        }
+        catch (int err_val)
+        {
+          switch(err_val)
+          {
+            case -1:
+              ROS_WARN_STREAM("Invalid file name entered; Deactivate fault and try again.");
+              break;
+            case -2:
+              ROS_WARN_STREAM("Could not find CSV in the 'profiles' directory with the provided name. Deactivate fault and try again.");
+              break;
+            default:
+              break;
+          }
+          custom_fault_ready = false;
+        }
+      }
+      else
+      {
+        // It's the same fault profile as previously stored, meaning the same fault was reactivated.
+        // Skip loading and continue from previous index.
+        ROS_INFO_STREAM(saved_file.substr(0, saved_file.size() - 4) << " re-activated!");
+        custom_fault_ready = true;
+      }
+    }
+    fault_activated = true;
+  }
+  else if (fault_activated && !fault_enabled)
+  {
+    ROS_INFO_STREAM("Custom power fault de-activated!");
```

Comment:
In this and all messages that refer to the "custom" fault, it would be more helpful to identify it by the profile name.  If this is hard, don't worry about it.